src/CSnakes.Runtime/CPython/Object.cs

```diff
@@ -154,4 +162,30 @@ internal static bool HasAttr(PyObject ob, string name)
     /// <returns></returns>
     [LibraryImport(PythonLibraryName)]
     internal static partial int PyObject_SetItem(PyObject ob, PyObject key, PyObject value);
+
+    [LibraryImport(PythonLibraryName)]
+    internal static partial int PyObject_Hash(PyObject ob);
+
+    internal static bool PyObject_Equals(PyObject ob1, PyObject ob2)
+    {
+        int result = PyObject_RichCompareBool(ob1, ob2, Py_EQ);
+        if (result == -1)
+        {
+            PyObject.ThrowPythonExceptionAsClrException();
+        }
+        return result == 1;
+    }
+
+    internal static bool PyObject_NotEquals(PyObject ob1, PyObject ob2)
+    {
+        int result = PyObject_RichCompareBool(ob1, ob2, Py_NE);
+        if (result == -1)
+        {
+            PyObject.ThrowPythonExceptionAsClrException();
+        }
+        return result == 1;
+    }
+
+    [LibraryImport(PythonLibraryName)]
+    internal static partial int PyObject_RichCompareBool(PyObject ob1, PyObject ob2, int opid);

```

Comment:
```suggestion
    internal static partial int PyObject_RichCompareBool(PyObject ob1, PyObject ob2, [MarshalAs(UnmanagedType.I4)] CompareType opid);
```

You can then turn the list of constants into an `enum`.